openwisp_controller/subnet_division/rule_types/device.py

```diff
@@ -0,0 +1,37 @@
+from django.db.models.signals import post_delete
+from swapper import load_model
+
+from ...config.signals import device_registered
+from .base import BaseSubnetDivisionRuleType
+
+Subnet = load_model('openwisp_ipam', 'Subnet')
+
+
+class DeviceSubnetDivisionRuleType(BaseSubnetDivisionRuleType):
+    provision_signal = device_registered
```

Comment:
we may have to use the classic post save signal (when config is created) to make it more robust (works also when devices are created differently).

We should make it more explicit that this rule works with Device Configurations (not just Devices)